client/my-sites/site-indicator/site-indicator.jsx

```diff
@@ -236,14 +204,16 @@ class SiteIndicator extends Component {
 		);
 	}
 
-	makeAnalyticsRecordEventHandler = action => {
-		return () => {
-			analytics.ga.recordEvent( 'Site-Indicator', action );
-		};
+	trackSiteDisconnect = () => {
+		this.props.recordGoogleEvent(
+			'Jetpack',
+			'Clicked in site indicator to start Jetpack Disconnect flow'
+		);
+		this.props.recordTracksEvent( 'calypso_jetpack_site_indicator_disconnect_start', {
+			siteId: this.props.site.ID,
+		} );
```

Comment:
Not blocking, but this is a candidate to use `composeAnalytics`
```jsx
const mapDispatchToProps = dispatch => ( {
	trackSiteDisconnect: ( siteId ) =>
		dispatch(
			composeAnalytics(
				recordGoogleEvent( 'Jetpack', 'Clicked in site indicator to start Jetpack Disconnect flow' ),
				recordTracksEvent( 'calypso_jetpack_site_indicator_disconnect_start', siteId )
			)
		),
} );
```

Reply:
I tried to play with it for a while, trying to avoid rebinding the action creators on every props change (which I believe the `( dispatch, { site } )` form would do), and finally decided to leave `trackSiteDisconnect` as it is 🙂

Reply:
Would this still cause the same trouble? 
```jsx
const mapDispatchToProps = ( dispatch ) => ( {
	trackSiteDisconnect: ( siteId ) =>
		dispatch(
			composeAnalytics(
				recordGoogleEvent( 'Jetpack', 'Clicked in site indicator to start Jetpack Disconnect flow' ),
				recordTracksEvent( 'calypso_jetpack_site_indicator_disconnect_start', siteId )
			)
		),
} );
```

Reply:
With an action creator like this, I still need to have a handler method:
```js
render() {
  return <WPAdminLink onClick={ this.trackSiteDisconnect } />
}

trackSiteDisconnect = () => this.props.trackSiteDisconnect( this.props.site.ID )
```
It's certainly possible to make `composeAnalytics` work here, but the code doesn't get any simpler IMHO.

Reply:
So, @seear already implemented this `composeAnalytics` suggestion in #22806 😄 And because he had the awesome observation that the `siteId` param is not needed, the code gets much simpler. After rebasing, the commit that fixed `trackSiteDisconnect` disappeared from this PR.